go-controller/pkg/libovsdbops/switch.go

```diff
@@ -348,21 +349,22 @@ type logicalSwitchPortPredicate func(*nbdb.LogicalSwitchPort) bool
 func DeleteLogicalSwitchPortsWithPredicateOps(nbClient libovsdbclient.Client, ops []libovsdb.Operation, sw *nbdb.LogicalSwitch, p logicalSwitchPortPredicate) ([]libovsdb.Operation, error) {
 	sw, err := GetLogicalSwitch(nbClient, sw)
 	if err != nil {
-		return nil, fmt.Errorf("error getting switch %+v: %v", sw, err)
+		if errors.Is(err, libovsdbclient.ErrNotFound) {
+			return ops, nil
+		} else {
+			return ops, fmt.Errorf("error getting switch %+v: %w", sw, err)
+		}
 	}
 
-	lsps := []*nbdb.LogicalSwitchPort{}
+	var lsps []*nbdb.LogicalSwitchPort
 	for _, port := range sw.Ports {
 		lsp := &nbdb.LogicalSwitchPort{UUID: port}
-		lsp, err := GetLogicalSwitchPort(nbClient, lsp)
+		lsp, err = GetLogicalSwitchPort(nbClient, lsp)
 		if err != nil {
-			return nil, fmt.Errorf("error getting port %+v: %v", lsp, err)
+			return ops, fmt.Errorf("error getting port %+v: %w", lsp, err)
```

Comment:
We would be usually be returning nil as ops when there is an error, curious about the change.

Reply:
I wanted to stay consistent within this func - I am returning the error message wraped so that the caller can decide how to proceed and also the ops so its convenient for them to proceed if they chose to do so. I  reverted this if its not the norm within the `libovsdbops` pkgs.